Test cleanup

elken opened the pull request:
- Made database.cpp a bit neater (Ideas for a better model on the way)
- Made tests idempotent
- Finalize CMake adjustment (just needs Mac-related things)
- Cleaned up src/CMakeLists.txt
- Fixed icons in DRM panes (due to path refactor)
- Adjusted HCI for DRM Wizard flow
- Cleaned up tests/CMakeLists.txt
- Removed gtest as a module
- Move to Qt Test
- Adjust CMakeLists to be more distributed (need to finalize commands and stuff for win/apple)
- Adjusted header paths
- Work on fixing the icon
- Added Travis builds for OSX
- Made (some) areas more C++11-y

Commit:
Test cleanup

Commit:
Comments

Commit:
Modified .gitignore

Commit:
Merge branch 'dev' of https://github.com/Proj-Ascension/Client into testCleanup

Commit:
Make qt plugins work

Commit:
Further fix CMake deploy (need to make a custom deploy target next)

Commit:
Fix me being an idiot

Commit:
Possible breaking changes here

Commit:
Submodule

Commit:
no message

Commit:
no message

Commit:
Conditional testing

Commit:
Decl issue

Commit:
Merge remote-tracking branch 'upstream/dev' into testCleanupTest

Commit:
Bunch of changes

- Removed gtest as a module
- Went back to Catch
- Adjust CMakeLists to be more distributed (need to finalize commands and stuff for win/apple)
- Adjusted header paths
- Work on fixing the icon

Commit:
Final CMake refactor (might break shit)

- Finalize CMake adjustment (just needs Mac-related things)
- Cleaned up src/CMakeLists.txt
- Fixed icons in DRM panes (due to path refactor)
- Adjusted HCI for DRM Wizard flow
- Cleaned up tests/CMakeLists.txt

Commit:
Merge branch 'testCleanupTest' of github.com:elken/client into testCleanupTest

Commit:
no message

Commit:
Merge branch 'testCleanupTest' of https://github.com/elken/Client into testCleanupTest

Commit:
no message

elken committed:
Merge pull request #4 from elken/testCleanupTest

Test cleanup test

Commit:
Fix linux path issue

Commit:
Merge branch 'elken-testCleanup' into dev

Commit:
Merge remote-tracking branch 'upstream/dev' into testCleanup

Hal9007 commented on src/drm/steam_drm.cpp:
```diff
@@ -30,18 +29,18 @@ void SteamDRM::checkExists()
+    }
 
-    if (steamFolder.filePath("").trimmed() != "" && steamFolder.exists() && steamFolder != QDir("."))
+    if (QDir(steamAppsDir.filePath("libraryFolders.vdf")).exists() && steamFolder.filePath("").trimmed() != "" && steamFolder.exists() && steamFolder != QDir("."))
```
Should be QFile.exists() in the first conditional here.

elken replied:
You're right, don't commit at 3am xD

On 3 February 2016 05:44:18 GMT+00:00, Jake Schadewe notifications@github.com wrote:

> > -    if (steamFolder.filePath("").trimmed() != "" &&
> >   steamFolder.exists() && steamFolder != QDir("."))
> > -    if (QDir(steamAppsDir.filePath("libraryFolders.vdf")).exists()
> >   && steamFolder.filePath("").trimmed() != "" && steamFolder.exists() &&
> >   steamFolder != QDir("."))
> 
> Should be QFile.exists() in the first conditional here.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/Proj-Ascension/Client/pull/118/files#r51680955

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

Commit:
Final test stuff and a few other fixes

- Hardcode output directories (will likely be changed but works for now)
- QDir -> QFile in steam_drm.cpp
- Added check for ResultsPage (caused the elements to get added multiple times before)
- Added a small check to verify what VdfParse is doing when it fails
- Moved to Qt Test, GUI tests coming soon™

Commit:
Merge branch 'testCleanup' of https://github.com/elken/Client into testCleanup

Commit:
Forgot travis.yml and AUTOMOC

Commit:
(Hopefully) Fixed drm lambdas

Commit:
Oops

Commit:
Linux fixes

Commit:
Qt version issue on travis

elken commented:
Once this is merged, I'll add a Travis webhook too.

Commit:
Supress warning about incorrect logic

Commit:
Fixed SHA test

Commit:
Fix dead space on *NIX

elken committed:
Update .travis.yml

Commit:
Revert source_group

Commit:
Merge branch 'testCleanup' of github.com:elken/client into testCleanup
